internal/host/host.go

```diff
@@ -351,15 +351,30 @@ func (m *Manager) updateInventory(ctx context.Context, cluster *common.Cluster,
 		return common.NewApiError(http.StatusInternalServerError, err)
 	}
 
-	// Remove virtual interfaces if the feature is not enabled
-	if !m.Config.EnableVirtualInterfaces {
-		var physicalInterfaces []*models.Interface = make([]*models.Interface, 0)
-		for _, intf := range inventory.Interfaces {
-			if intf.Type == "" || intf.Type == "physical" || intf.Type == "vlan" || intf.Type == "bond" {
-				// Empty interface type indicates an older agent, which only passes physical interfaces
-				physicalInterfaces = append(physicalInterfaces, intf)
-			}
+	existingHostInventory, err := common.UnmarshalInventory(h.Inventory)
```

Comment:
How about moving all this part into a function and place it under https://github.com/openshift/assisted-service/blob/9b5bb30c4e1c7ba42566f11eabb09fe16eaf3942/internal/bminventory/inventory.go#L2948 switch? 
There is a log of logic here that is not related to updateInventroy
@carbonin @CrystalChun what do you think?

Reply:
Agreed with Nick that the logic fits in this section since we're modifying/updating the interfaces (one aspect of the inventory which is similar to disk eligibility, etc.)

Reply:
OK